bin/main.py

```diff
@@ -0,0 +1,37 @@
+#!/usr/bin/env python3
+
+"""
+Usage:
+  main.py backup
+  main.py restore
+  main.py dump_ssh
+"""
+import shutil
+from os.path import join
+
+from docopt import docopt
+
+from backup import run_backup
+from logger import with_logging
+from restore import run_restore
+from settings import log_dir, ssh_key_path, known_hosts_path, root_path
+
+
+def run():
+    args = docopt(__doc__)
+    if args["backup"]:
+        print("Backing up targets to Starport. Output will be logged to " + log_dir)
+        run_backup()
+    elif args["restore"]:
+        print("Restoring targets from Starport. Output will be logged to " + log_dir)
+        run_restore()
+    elif args["dump_ssh"]:
+        print("Storing ssh key and known_hosts file in volume (mounted at /bb8/etc/.ssh)")
+        dest = join(root_path, ".ssh")
+        shutil.copy(ssh_key_path, join(dest, "id_rsa"))
```

Comment:
Looks like this root path is a path outside the docker container right? And this script is running inside the container. So don't think this works, unless I'm missing something.

Reply:
No, the way it works is like this:

1. In setup.sh we obtain_secrets.py and store them on the host machine
2. Then we build the docker image, copying the secrets in
3. Then, in main.py, when dump_secrets is invoked, we transform the secrets and store them in the .ssh volume.

So ssh_key_path in `/etc/bb8/secrets/ssh_key`, which is inside the container, copied in by Dockerfile.